velox/functions/lib/Hash.h

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cstdint>
+
+namespace facebook::velox::functions {
+/// Murmur3 aligns with Austin Appleby
+/// https://github.com/aappleby/smhasher/blob/master/src/MurmurHash3.cpp
+///
+/// Signed integer types have been remapped to unsigned types (as in the
+/// original) to avoid undefined signed integer overflow and sign extension.
+class Murmur3Hash32 {
+ public:
+  /// Hash the lower int , is a fast path of hashBytes.
+  static uint32_t hashInt64(uint64_t input, uint32_t seed);
+
+ protected:
+  static uint32_t mixK1(uint32_t k1);
```

Comment:
Are these the same as in spark? Let's remove mixK1, mixH1 and fmix duplicates from spark.